modules/core/src/lifecycle/props.js

```diff
@@ -173,6 +174,29 @@ function diffUpdateTriggers(props, oldProps) {
   return reason;
 }
 
+// Returns true if any extensions have changed
+function diffExtensions(props, oldProps) {
+  if (oldProps === null) {
+    return 'oldProps is null, initial diff';
+  }
+
+  const oldExtensions = oldProps.extensions;
+  const {extensions} = props;
+
+  if (extensions === oldExtensions) {
+    return false;
+  }
+  if (extensions.length !== oldExtensions.length) {
+    return true;
+  }
+  for (let i = 0; i < extensions.length; i++) {
+    if (!extensions[i].equals(oldExtensions[i])) {
```

Comment:
Probably not a major concern, but to we care about the case when the it's the same extensions in a different order?

Reply:
Theoretically order matters because extensions can overwrite the changes done by a previous extension... But that's another can of worms.